scripts/HighNoon.js

```diff
@@ -0,0 +1,401 @@
+    constructor(maxPlayers) {
+      this.players = {};
+      this.maxPlayers = maxPlayers;
+      this.shotsFired = false;
+      this.countdownValue = Math.floor(Math.random() * 2500) + 2500; // Guarantees to never be zero
+      this.timer = null;
+      this.elapsed = 0;
+    }
+
+    playersNeeded() {
+      return this.maxPlayers - this.players.length;
+    }
+
+    getCurrentPlayers() {
+      return this.players;
+    }
+
+    getMaxPlayers() {
+      return this.maxPlayers;
+    }
+
+    getCountDownValue() {
+      return this.countdownValue;
+    }
+
+    sessionIsFull() {
+      return this.players.length === this.maxPlayers;
+    }
+
+    incrementTimer() {
+      this.elapsed += 2500;
+    }
+
+    timeToDraw() {
+      return this.elapsed >= this.countDownValue;
+    }
+
+    duelIsDone() {
+      this.allShotsFired = true;
+    }
+
+  };
+
+  class Session {
+    constructor(channelId, maxPlayers) {
+      this.channelId = channelId;
+      this.gameState = new GameState(maxPlayers);
+    }
+
+    printDots(msg) {
+      msg.reply('...');
+      this.gameState.incrementTimer();
+    }
+
+    startTimer(msg) {
+      this.gameState.timer = setInterval(function() { printDots(msg); }, this.gameState.countdownValue);
+    }
+
+    clearTimer() {
+      clearInterval(this.gameState.timer);
+    }
+
+    printDraw(msg, timer) {
+      msg.reply('DRAW!');
+      clearTimer();
+    }
+
+    addPlayer(person) {
+      if (!this.gameState.sessionIsFull()) {
+        if (this.gameState.players[person] === undefined) {
+          this.gameState.players[person] = new PlayerState(person);
+          return PLAYER_ADDED;
+        } else {
+          return PLAYER_ALREADY_IN_SESSION;
+        }
+      } else {
+        return SESSION_FULL;
+      }
+    }
+
+    removePlayer(person) {
+      if (this.gameState.players[person] !== undefined) {
+        delete this.gameState.players[person];
+        return PLAYED_REMOVED;
+      } else {
+        return PLAYER_DOES_NOT_EXIST;
+      }
+    }
+
+  };
+
+  ////////////// SESSION METHODS //////////////
+
+  var createSession = function (channelId, person, maxPlayers = 2) {
+    if (sessionStore[channelId]) {
+      return sessionStore[channelId].addPlayer(person);
+    } else {
+      sessionStore[channelId] = new Session(channelId, maxPlayers);
+      sessionStore[channelId].addPlayer(person);
+      return SESSION_CREATED;
+    }
+  };
+
+  var removeSession = function (msg, channelId) {
+    var existed = Boolean(sessionStore[channelId]); // check if it exists first prior to deletion
+    if (!existed) {
+      msg.reply('There was not a session made yet!  Please type &highnoon to start a session.');
+    } else {
+      delete sessionStore[channelId];
+      msg.reply('Session successfully removed from room ' + channelId);
+    }
+  };
+
+  ////////////// LISTENERS //////////////
+
+  robot.hear(trigger.getTrigger(), function (msg) {
+    var person = messageHelper.getPerson(msg);
```

Comment:
You do a lot of logic here in these listeners.  You should keep your listeners very short and move much of their logic into your `GameEngine` classes.